### PR TITLE
feat: [CI-4717]: Support for Bitbucket PR Comment triggers

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34680)

### CODE REVIEW

**Review by jamie-harness — APPROVED** [review]
> Reviewed 5 of 11 files at r1, 8 of 9 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34680)* status: :shipit: complete! ✓

File: 810-ng-triggers/src/main/java/io/harness/ngtriggers/beans/source/webhook/v2/bitbucket/event/BitbucketPRCommentSpec.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import io.harness.ngtriggers.beans.source.webhook.v2.git.GitEvent;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.experimental.FieldDefaults;
+
+@Data
```
Comment (harshjain12) [inline 810-ng-triggers/src/main/java/io/harness/ngtriggers/beans/source/webhook/v2/bitbucket/event/BitbucketPRCommentSpec.java:30]: @Value

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 3 of 11 files at r1, 6 of 9 files at r2, 3 of 4 files at r3, 2 of 2 files at r4, 7 of 7 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34680#-:-N5dexx84fqVeJQiDiFD:bnfp4nl)* status: :shipit: complete! ✓

**Review by jamie-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r4, 1 of 7 files at r5, 1 of 4 files at r6, 1 of 7 files at r7, 10 of 15 files at r8, 4 of 4 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34680)* status: 2. unresolved discussions.

**Review by rutvijmehta-harness — COMMENTED** [review]
> Reviewed 1 of 9 files at r2, 1 of 4 files at r3, 1 of 2 files at r4, 1 of 7 files at r5, 1 of 4 files at r6, 1 of 7 files at r7, 10 of 15 files at r8, 4 of 4 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34680)* status: 2. unresolved discussions.

**Pull request merged**